Applications/FileManager/main.cpp

```diff
@@ -649,8 +649,23 @@ int run_in_windowed_mode(RefPtr<Core::ConfigFile> config, String initial_locatio
             tree_view.update();
         }
 
-        go_forward_action->set_enabled(directory_view.path_history_position()
-            < directory_view.path_history_size() - 1);
+        struct stat st;
+        if (lstat(new_path.characters(), &st)) {
+            perror("stat");
+            return;
+        }
+
+        bool can_write_in_path = false;
+        if (st.st_uid == getuid() && st.st_mode & S_IWUSR) {
+            can_write_in_path = true;
+        } else if (st.st_uid != getuid() && st.st_gid == getgid() && st.st_mode & S_IWGRP) {
+            can_write_in_path = true;
+        } else if (st.st_uid != getuid() && st.st_gid != getgid() && st.st_mode & S_IWOTH) {
+            can_write_in_path = true;
+        }
```

Comment:
Something like this would make a great addition to the standard library. It would be useful in the Terminal for suggesting executables.

Reply:
Yes, I thought about that. I was thinking if I could move this to some place on the standard library and use to also check if we can paste or delete and enable/disable the button accordingly...

I'll go to IRC and ask, I suppose is better.

Reply:
The kernel already provides this functionality via the `access` syscall :)

Reply:
Learned something new today